In [14]:
import pandas as pd
import numpy as np
import matplotlib 
import seaborn
import plotly.express as px

In [2]:
Tik_comments = pd.read_csv("final_data.csv") 

C:\Users\luke gevas\AppData\Local\Temp\ipykernel_21648\1428500902.py:1: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  Tik_comments = pd.read_csv("final_data.csv")


I will check the first five rows of data ao i know what I am working with.

In [3]:
Tik_comments.head(5)

comment_id                                               text  \
0  6816879848683438085  But are you giving him a Tesla? Because he des...   
1  6816846992812015622              David was just hoping for some help 😢   
2  6816942564911628294  We ain’t forgot that time he tricked us all in...   
3  6816824764519301125                                    get him a tesla   
4  6816821633249001477                                   david ur so hung   

              video_id  create_time  like_count  status  author.unique_id  \
0  6816821384652557574   1587178528         161       1      lissettecalv   
1  6816821384652557574   1587170875        1165       1  hellomyfriend0_o   
2  6816821384652557574   1587193131         249       1     briancarrion_   
3  6816821384652557574   1587165700        4038       1      slumpgodkayy   
4  6816821384652557574   1587164971         100       1             _4den   

               author.nickname  author.is_private author.language  \
0  Lissette | Influencer Coach              False              en   
1               Isaac micallef               True              en   
2                 Briancarrion              False              en   
3          kaykay the slumpgod              False              en   
4                          dad              False              en   

                                    author.signature author.custom_verify  \
0  How to BUILD & MONETIZE a digital biz with pur...     Verified account   
1                   In bed watching Tiktok video's 😴                  NaN   
2                    Might make you cringe 🤷🏽‍♂️\n🇻🇪     Verified account   
3            👻:@ kaykayyishere\nfollow my instagram💖                  NaN   
4                                   x kinda fell off                  NaN   

            author.uid                                     author.sec_uid  \
0   133202372536832000  MS4wLjABAAAAbX7SywQS04x1ilOoKHrpTZqhQ6lD32nRC6...   
1  6744596691107824645  MS4wLjABAAAADslkcxnt1mqt2oQwQEK8d0ybBHPuCO-geV...   
2              2165547  MS4wLjABAAAAKH09hef-gLcTyrzYygfGHMazAhFaLMML7-...   
3  6586423495932313605  MS4wLjABAAAAI1dyIIb3qlBf3LhqS3-OJITdYuc3OcWlsd...   
4  6740418744587650054  MS4wLjABAAAAykXQ0pVI4w-y3kI1omvYzOH2xnJN0iP3lf...   

                                 author.avatar_thumb author.region  \
0  https://p77-sign-sg.tiktokcdn.com/musically-ma...            US   
1  https://p77-sign-sg.tiktokcdn.com/musically-ma...            MT   
2  https://p16-sign-sg.tiktokcdn.com/musically-ma...            US   
3  https://p16-sign-sg.tiktokcdn.com/musically-ma...            US   
4  https://p77-sign-sg.tiktokcdn.com/tos-maliva-a...            US   

    author.ins_id author.youtube_channel_title author.youtube_channel_id  \
0    lissettecalv                          NaN                       NaN   
1             NaN                          NaN                       NaN   
2  brian.carrion_                          NaN                       NaN   
3     kaylemraley                          NaN                       NaN   
4           _4den                 aden burgess  UCe3pSCjGls_hByctWn91rqw   

   author.twitter_id  
0                NaN  
1                NaN  
2                NaN  
3                NaN  
4                NaN

In [4]:
average_like_count = Tik_comments["like_count"].mean()
average_like_count

74.97258797165526

The information above gives me the average number of likes each person has on their own TikTok account. 75 tells me that a few of his commentors might be influencers, but most are probably people who don't regularly create their own content on TikTok

In [5]:
Tik_comments.isnull().sum()

comment_id                           0
text                                 0
video_id                             0
create_time                          0
like_count                           0
status                               0
author.unique_id                     0
author.nickname                      0
author.is_private                    0
author.language                      0
author.signature                 38878
author.custom_verify            164364
author.uid                           0
author.sec_uid                       0
author.avatar_thumb                  0
author.region                      135
author.ins_id                   115000
author.youtube_channel_title    145634
author.youtube_channel_id       145634
author.twitter_id               165012
dtype: int64

Now I know which columns in my data are actually missing information. And I can begin to clean it. I don't want to drop anything as I think it would be interesting to examine all these columns further

In [6]:
Tik_comments["author.youtube_channel_title"].fillna("No Channel", inplace = True) 
Tik_comments["author.signature"].fillna("False", inplace=True)
Tik_comments["author.custom_verify"].fillna("False", inplace=True)
Tik_comments["author.region"].fillna("False", inplace=True)
Tik_comments["author.ins_id"].fillna("False", inplace=True)
Tik_comments["author.youtube_channel_id"].fillna("False", inplace=True)
Tik_comments["author.twitter_id"].fillna("False", inplace=True)

In [7]:
Tik_comments.isnull().sum()

comment_id                      0
text                            0
video_id                        0
create_time                     0
like_count                      0
status                          0
author.unique_id                0
author.nickname                 0
author.is_private               0
author.language                 0
author.signature                0
author.custom_verify            0
author.uid                      0
author.sec_uid                  0
author.avatar_thumb             0
author.region                   0
author.ins_id                   0
author.youtube_channel_title    0
author.youtube_channel_id       0
author.twitter_id               0
dtype: int64

In [8]:
Tik_comments.video_id

0         6816821384652557574
1         6816821384652557574
2         6816821384652557574
3         6816821384652557574
4         6816821384652557574
                 ...         
165105    6831297085137095942
165106    6831297085137095942
165107    6831297085137095942
165108    6831297085137095942
165109    6831297085137095942
Name: video_id, Length: 165110, dtype: int64

I haven't decided what I want to do with this data. 
1. The amount of people that like his videos that all have their own youtube channel 
2. The amount of people who are private vs. public 
3. The region of the world his views come from 

In [9]:
youtubers = Tik_comments.loc[Tik_comments["author.youtube_channel_title"] != "No Channel"]
youtubers

comment_id  \
4       6816821633249001477   
13      6816874886972571654   
23      6816839795353026565   
24      6816942164397260805   
30      6816860870749339654   
...                     ...   
165081  6831301420651331589   
165086  6831298433941716998   
165095  6831297606040584197   
165104  6831297442744074245   
165106  6831297361822711814   

                                                     text  \
4                                        david ur so hung   
13                                        Best thing ever   
23      This two hours that I spend doing a david vide...   
24                           LMFAOOOO IM LAUGHING SO HARD   
30      If u see a cute lil silver Acura driving aroun...   
...                                                   ...   
165081                                                wow   
165086                            Seems like a fair trade   
165095                                     Does he reply?   
165104                             Pls reply! Ur my idol!   
165106                                                    

                   video_id  create_time  like_count  status  \
4       6816821384652557574   1587164971         100       1   
13      6816821384652557574   1587177371         278       1   
23      6816821384652557574   1587169200          26       1   
24      6816821384652557574   1587193036         392       1   
30      6816821384652557574   1587174107          19       1   
...                     ...          ...         ...     ...   
165081  6831297085137095942   1590536310           8       1   
165086  6831297085137095942   1590535614           8       1   
165095  6831297085137095942   1590535421           8       1   
165104  6831297085137095942   1590535384           8       1   
165106  6831297085137095942   1590535364           8       1   

                author.unique_id    author.nickname  author.is_private  \
4                          _4den                dad              False   
13                    sam..frank      $ 𝓐 𝙼 𝔸 🄽 𝓣𝕳𝐀              False   
23                   karimsalmen       Karim Salmen              False   
24                        angzel             Angzel              False   
30                       .reilyn    REILYN GAGLIONE              False   
...                          ...                ...                ...   
165081               emily.ducky              Emily              False   
165086  insane.fortnite.glitches  Skin Code: Danger               True   
165095            vel_on_youtube                VEL              False   
165104              jonasbraedon              jonas              False   
165106         bartaintnosimpson             Swervz              False   

       author.language                                   author.signature  \
4                   en                                   x kinda fell off   
13                  en          ☯️✝️\nsamfrank306@gmail.com\nDm on IG\n👇🏼   
23                  en           Follow my instagram and youtube!\n🇻🇪🇺🇸🇱🇧   
24                  en  nyc 🖤🧝🏻‍♀️🔪\ndepop: @angelzap\ninsta: @angzel\...   
30                  en                        Don’t b shy follow my insta   
...                ...                                                ...   
165081              en      i like ducks :)\ni comment duck on every post   
165086              en  My socials ⬆️\nUse code Danger and follow my y...   
165095              en  YouTube: VEL\nInstagram: youtube_vel\nBroken F...   
165104              en        follow my insta\nbe happy ☻\n tysm for 2.6k   
165106              en                                              16 🇵🇱   

       author.custom_verify           author.uid  \
4                     False  6740418744587650054   
13                    False  6628277104105586693   
23                    False   257637576054820864   
24                    False  6774494504173241349   
30                    False  6662497387686608902   
...             

In [11]:
youtubers.nunique()

comment_id                       849
text                             831
video_id                           5
create_time                      843
like_count                       363
status                             1
author.unique_id                 806
author.nickname                  803
author.is_private                  2
author.language                   10
author.signature                 776
author.custom_verify               4
author.uid                       806
author.sec_uid                   806
author.avatar_thumb             1612
author.region                     43
author.ins_id                    567
author.youtube_channel_title     804
author.youtube_channel_id        805
author.twitter_id                  5
dtype: int64

In [10]:
youtubers_unique = Tik_comments.drop_duplicates(subset=["author.youtube_channel_title"], keep="first")
youtubers_unique

comment_id  \
0       6816879848683438085   
4       6816821633249001477   
13      6816874886972571654   
23      6816839795353026565   
24      6816942164397260805   
...                     ...   
144436  6831297785820659717   
144444  6831436466107858949   
144493  6831371342508687366   
144494  6831367984108781574   
144498  6831355292098363398   

                                                     text  \
0       But are you giving him a Tesla? Because he des...   
4                                        david ur so hung   
13                                        Best thing ever   
23      This two hours that I spend doing a david vide...   
24                           LMFAOOOO IM LAUGHING SO HARD   
...                                                   ...   
144436  david do you know what happend with @brycehall...   
144444          I'm framing that dollar, you tripping lol   
144493                                         @sky_clown   
144494                                          Fantastic   
144498                          I live in Tampa wya David   

                   video_id  create_time  like_count  status  \
0       6816821384652557574   1587178528         161       1   
4       6816821384652557574   1587164971         100       1   
13      6816821384652557574   1587177371         278       1   
23      6816821384652557574   1587169200          26       1   
24      6816821384652557574   1587193036         392       1   
...                     ...          ...         ...     ...   
144436  6831297085137095942   1590535463          11       1   
144444  6831297085137095942   1590567752          10       1   
144493  6831297085137095942   1590552590           8       1   
144494  6831297085137095942   1590551808           8       1   
144498  6831297085137095942   1590548852           8       1   

            author.unique_id              author.nickname  author.is_private  \
0               lissettecalv  Lissette | Influencer Coach              False   
4                      _4den                          dad              False   
13                sam..frank                $ 𝓐 𝙼 𝔸 🄽 𝓣𝕳𝐀              False   
23               karimsalmen                 Karim Salmen              False   
24                    angzel                       Angzel              False   
...                      ...                          ...                ...   
144436          star_m00n.12                xxxtentacion❤              False   
144444              sticqeno                      Jáy-Qué              False   
144493           sydkneejade                    Sydney ✔️              False   
144494              luverri8                      Cesar ☽              False   
144498  basicallyjustjohnnyy                     Johnny 🦋              False   

       author.language                                   author.signature  \
0                   en  How to BUILD & MONETIZE a digital biz with pur...   
4                   en                                   x kinda fell off   
13                  en          ☯️✝️\nsamfrank306@gmail.com\nDm on IG\n👇🏼   
23                  en           Follow my instagram and youtube!\n🇻🇪🇺🇸🇱🇧   
24                  en  nyc 🖤🧝🏻‍♀️🔪\ndepop: @angelzap\ninsta: @angzel\...   
...                ...                                                ...   
144436              en                                                 hi   
144444              en                I stream. \n👇🏾Follow me on twitch👇🏾   
144493              en  FOLLOW ME ON \n🖤Instagram  ➪  @Syd.knee.jade\n...   
144494              en      🎴\nSc: cesarberrioss\nIg: luverri8\nFollow me   
144498              en                         💜👉🏼👈🏼🥺\nTwitter: @johnawho   

       author.custom_verify           author.uid  \
0          Verified account   133202372536832000   
4                     False  6740418744587650054   
13                    False  6628277104105586693   
23                    False   257637576054820864

805 different youtubers have left comments on David's tiktok! 

In [17]:
fig1 = px.histogram(Tik_comments, x="author.region")
fig1.show()